plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -382,6 +382,116 @@ HAVING
 OPTION(RECOMPILE);
 `
 
+const sqlAzureDBPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+DECLARE @lastIntervalEndTimestamp BIGINT = 0;
+
+IF @QueryData != '' AND @QueryData IS NOT NULL
+	SET @lastIntervalEndTimestamp = CAST(@QueryData AS BIGINT);
+
+DECLARE @magicNumberDays BIGINT = 4294967296, @magicNumberSeconds BIGINT = 300, @secondsInDay INT = 86400;
+
+DECLARE @timeGrain BIGINT = (SELECT interval_length_minutes * 60 FROM sys.database_query_store_options WITH (NOLOCK)); --use query store time grain
+
+DECLARE @currDate DATETIME = GETUTCDATE();
+DECLARE @currIntervalEndTimestamp BIGINT = CONVERT(BIGINT, DATEDIFF(day, 0, @currDate)) * @secondsInDay + DATEDIFF(second, DATEADD(day, DATEDIFF(day, 0, @currDate), 0), @currDate) / @timeGrain * @timeGrain;
+IF @lastIntervalEndTimestamp = @currIntervalEndTimestamp
+	RETURN;
+
+DECLARE @currIntervalStartTimestamp BIGINT = @currIntervalEndTimestamp - @timeGrain;
+IF @lastIntervalEndTimestamp != 0
+	SET @currIntervalStartTimestamp = @lastIntervalEndTimestamp;
+
+DECLARE @currIntervalEndDate datetime = DATEADD(second, @currIntervalEndTimestamp % @secondsInDay, DATEADD(day, @currIntervalEndTimestamp / @secondsInDay, 0));
+DECLARE @currIntervalStartDate datetime = DATEADD(second, @currIntervalStartTimestamp % @secondsInDay, DATEADD(day, @currIntervalStartTimestamp / @secondsInDay, 0));
+
+`
+
+const sqlAzureDBQueryStoreRuntimeStatistics = sqlAzureDBPartQueryPeriod + `
+SET DEADLOCK_PRIORITY -10;
+IF SERVERPROPERTY('EngineEdition') = 5  -- Is this Azure SQL DB?
+BEGIN
+SELECT
+	'sqlserver_azuredb_querystore_runtime_stats' AS [measurement],
+	CONVERT(varchar(36), NEWID()) AS [measurement_id],
+	REPLACE(@@SERVERNAME, '\', ':') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(datetime, rsi.start_time, 1) AS interval_start_time,
+	CONVERT(datetime, rsi.end_time, 1) AS interval_end_time,
+	CONVERT(BIGINT, DATEDIFF(day, 0, CONVERT(datetime, rsi.start_time, 1))) * @magicNumberDays + DATEDIFF(second, 0, DATEADD(day,-DATEDIFF(day, 0, CONVERT(datetime, rsi.start_time, 1)), CONVERT(datetime, rsi.start_time, 1))) * @magicNumberSeconds AS interval_start_time_d,
+	CONVERT(BIGINT, DATEDIFF(day, 0, CONVERT(datetime, rsi.end_time, 1))) * @magicNumberDays + DATEDIFF(second, 0, DATEADD(day,-DATEDIFF(day, 0, CONVERT(datetime, rsi.end_time, 1)), CONVERT(datetime, rsi.end_time, 1))) * @magicNumberSeconds AS interval_end_time_d,
+	q.query_id,
+	CONVERT(nvarchar(max), q.query_hash, 1) as query_hash,
+	p.plan_id,
+	CONVERT(nvarchar(max), q.last_compile_batch_sql_handle, 1) as last_compile_batch_sql_handle,
```

Comment:
Why do we need last_compile_batch_sql_handle?

Reply:
Removed